pandas/core/internals/array_manager.py

```diff
@@ -473,15 +509,19 @@ def is_single_block(self) -> bool:
 
     def get_bool_data(self, copy: bool = False) -> ArrayManager:
```

Comment:
Could share this whole method with BM with something like:

```
out = []
indices = []

for arr, locs in enumerate(self):
    if is_bool_dtype(arr.dtype) or (is_object_dtype(arr.dtype) and lib.is_bool_array(arr.ravel("K"))):
         [...]

def __iter__(self): # <-- AM
    for i, arr in enumerate(self.arrays):
        yield i, arr

def __iter__(self):  # <-- BM
    for blk in self.blocks:
        yield blk.values, blk.mgr_locs
```

Reply:
The BlockManager uses different logic to determine if a Block is boolean or not, i.e. `Block.is_bool` (and for this case also splits up object blocks, which it doesn't do for get_numeric_data) 
(so with the current implementation, there is not much to share)

Reply:
OK, thanks for taking a look